### PR TITLE
Add key checks

### LINKED ISSUE #131: Need a way to check if a key is available
## Description
If someone needs to check if galois_keys are present in a context, he will need to do that in a try except since accessing that attribute will throw an error if the keys doesn't exist. We need a flexible way to check a boolean or something to know if a key is available )

## Additional Context

Here is how the error looks in the interpreter, where ctx is a TenSEALContext

```python
>>> ctx.galois_keys()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: the current context doesn't hold a Galois keys
```

### PR DESCRIPTION
**Pull request by IanQS**
## Description

Add key checks. Closes #131 

## Affected Dependencies

None

## How has this been tested?

Not tested but it seems fairly straightforward? 

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [X] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

File: tenseal/cpp/context/tensealcontext.cpp [hunk 1]
```diff
@@ -118,9 +118,22 @@ shared_ptr<TenSEALContext> TenSEALContext::Create(
     return shared_ptr<TenSEALContext>(new TenSEALContext(input, n_threads));
 }
 
+bool TenSEALContext::has_public_key() const {
+    if (this->_public_key == nullptr){
```
Comment (bcebere) [inline tenseal/cpp/context/tensealcontext.cpp:122]: why not 
```return this->_public_key != nullptr;```

File: tenseal/cpp/context/tensealcontext.cpp [hunk 2]
```diff
@@ -118,9 +118,22 @@ shared_ptr<TenSEALContext> TenSEALContext::Create(
...
+        return false;
+    }
+    return true;
+}
+
 shared_ptr<PublicKey> TenSEALContext::public_key() const {
     return this->_public_key;
 }
+bool TenSEALContext::has_secret_key() const {
+    if (this->_secret_key == nullptr){
```
Comment (bcebere) [inline tenseal/cpp/context/tensealcontext.cpp:132]: same here

File: tenseal/cpp/context/tensealcontext.cpp [hunk 3]
```diff
@@ -130,6 +143,12 @@ shared_ptr<SecretKey> TenSEALContext::secret_key() const {
     return this->_secret_key;
 }
 
+bool TenSEALContext::has_relin_keys() const {
+    if (this->_relin_keys == nullptr){
```
Comment (bcebere) [inline tenseal/cpp/context/tensealcontext.cpp:147]: same

File: tenseal/cpp/context/tensealcontext.cpp [hunk 4]
```diff
@@ -139,6 +158,13 @@ shared_ptr<RelinKeys> TenSEALContext::relin_keys() const {
     return this->_relin_keys;
 }
 
+bool TenSEALContext::has_galois_key() const {
+    if (this->_galois_keys == nullptr){
```
Comment (bcebere) [inline tenseal/cpp/context/tensealcontext.cpp:162]: same

**Comment by IanQS** [comment]
@youben11 Can I merge this?

File: tenseal/cpp/context/tensealcontext.cpp [hunk 5]
```diff
@@ -118,9 +118,16 @@ shared_ptr<TenSEALContext> TenSEALContext::Create(
     return shared_ptr<TenSEALContext>(new TenSEALContext(input, n_threads));
 }
 
+bool TenSEALContext::has_public_key() const {
+    return !(this->_public_key == nullptr);
```
Comment (philomath213) [inline tenseal/cpp/context/tensealcontext.cpp:122]: I think `return this->_public_key != nullptr;` is more readable.
same for others.

File: tenseal/cpp/context/tensealcontext.cpp [hunk 6]
```diff
@@ -130,6 +137,10 @@ shared_ptr<SecretKey> TenSEALContext::secret_key() const {
     return this->_secret_key;
 }
 
+bool TenSEALContext::has_relin_keys() const {
+    return !(this->_relin_keys != nullptr);
```
Comment (youben11) [inline tenseal/cpp/context/tensealcontext.cpp:141]: I think you left the outer ! by error

**Pull request merged**